07.ls_object/ls_directory.rb

```diff
@@ -23,10 +23,15 @@ def prepare_ls_files
     directory_files = Dir.foreach(@directory_path).to_a
     filtered_files = @hidden_file_presence ? directory_files : exclude_hidden_files(directory_files)
     ordered_files = @reversed_order ? filtered_files.sort.reverse : filtered_files.sort
-    ordered_files.map { |file| LsFile.new(@directory_path, file) }
+    ordered_files.map { |file| generate_ls_file(file) }
```

Comment:
> 一行に詰め込むのはどうなのかなと思い、その部分を別メソッドに切り出しました。

はい、「1行に詰め込みすぎない」という観点はとても良いと思います 👍 

ただし、メソッド化するのも一手ですが、わざわざメソッド化しなくてもこんな感じでブロックの中で全部書いてしまっても可読性は大きく変わらないと思います〜。

```suggestion
    ordered_files.map do |file| 
      file_path = File.join(@directory_path, file)
      LsFile.new(file_path)
    end
```

Reply:
ブロックの中で書くように変更しました。